src/Symfony/Component/DependencyInjection/Attribute/AutowireInline.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+
+/**
+ * Attribute to tell a parameter to inline a service with custom parameters.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireInline extends Autowire
+{
+    public function __construct(string|array $class, array $arguments = [], array $calls = [], array $properties = [], string $parent = null, bool|string $lazy = false)
+    {
+        parent::__construct([
+            \is_array($class) ? 'factory' : 'class' => $class,
+            'arguments' => $arguments,
+            'calls' => $calls,
+            'properties' => $properties,
+            'parent' => $parent,
```

Comment:
Could be nice to support but won't work as is because `ResolveChildDefinitionPass` already when this is discovered.